macros/edr/data_monitoring/monitors_query/table_monitoring_query.sql

```diff
@@ -1,78 +1,33 @@
             where
                 {{ elementary.cast_as_timestamp(timestamp_column) }} >= (select min(edr_bucket_start) from buckets)
                 and {{ elementary.cast_as_timestamp(timestamp_column) }} < (select max(edr_bucket_end) from buckets)
-                {% if where_expression %} and {{ where_expression }} {% endif %}
-        ),
-
-        {%- if 'row_count' in table_monitors %}
-
-        row_count_values as (
-            select edr_bucket_start,
-                   edr_bucket_end,
-                   start_bucket_in_data,
-                   case when start_bucket_in_data is null then
-                       0
-                   else {{ elementary.cast_as_float(elementary.row_count()) }} end as row_count_value
-            from buckets left join filtered_monitored_table on (edr_bucket_start = start_bucket_in_data)
-            group by 1,2,3
-        ),
-
-        row_count as (
-            select edr_bucket_start,
-                   edr_bucket_end,
-                   {{ elementary.const_as_string('row_count') }} as metric_name,
-                   {{ elementary.null_string() }} as source_value,
-                   row_count_value as metric_value
-            from row_count_values
-        ),
-
-        {%- else %}
-
-        row_count as (
-            {{ elementary.empty_table([('edr_bucket_start','timestamp'),('edr_bucket_end','timestamp'),('metric_name','string'),('source_value','string'),('metric_value','int')]) }}
-        ),
-
-        {%- endif %}
-
-        table_freshness as (
-        {%- if 'freshness' in table_monitors %}
-            {%- if freshness_column is undefined or freshness_column is none %}
-                {%- set freshness_column = timestamp_column %}
-            {%- endif %}
-            select
-                edr_bucket_start,
-                edr_bucket_end,
-                {{ elementary.const_as_string('freshness') }} as metric_name,
-                {{ elementary.cast_as_string('max('~freshness_column~')') }} as source_value,
-                {{ elementary.timediff('second', elementary.cast_as_timestamp('max('~freshness_column~')'), "edr_bucket_end") }} as metric_value
-            from buckets, {{ monitored_table_relation }}
-            where {{ elementary.cast_as_timestamp(timestamp_column) }} < edr_bucket_end
-            group by 1,2,3
-        {%- else %}
-            {{ elementary.empty_table([('edr_bucket_start','timestamp'),('edr_bucket_end','timestamp'),('metric_name','string'),('source_value','string'),('metric_value','int')]) }}
-        {%- endif %}
         ),
 
-        union_metrics as (
-
-        select * from row_count
-        union all
-        select * from table_freshness
-
+        metrics as (
+            {{ elementary.get_unified_metrics_query(metrics=table_monitors,
+                                                    metric_args=metric_args,
+                                                    timestamp_column=timestamp_column) }}
```

Comment:
I did a bit of a refactor here - instead of putting the queries for the different metrics in the body of the function, each is implemented in its own macro below and `get_unified_metrics_query` returns the union of them.

I felt it's needed since there is an additional metric, and the existing freshness metric has a more complex query now.